python/fusion_engine_client/messages/signal_defs.py

```diff
@@ -34,10 +35,87 @@ class SatelliteType(IntEnum):
 SatelliteTypeCharReverse = {v: k for k, v in SatelliteTypeChar.items()}
 
 
+class SatelliteTypeMask(IntEnum):
+    GPS = (1 << SatelliteType.GPS)
+    GLONASS = (1 << SatelliteType.GLONASS)
+    LEO = (1 << SatelliteType.LEO)
+    GALILEO = (1 << SatelliteType.GALILEO)
+    BEIDOU = (1 << SatelliteType.BEIDOU)
+    QZSS = (1 << SatelliteType.QZSS)
+    MIXED = (1 << SatelliteType.MIXED)
+    SBAS = (1 << SatelliteType.SBAS)
+    IRNSS = (1 << SatelliteType.IRNSS)
+
+    ALL = 0xFFFFFFFF
+
+    @classmethod
+    def to_bit_mask(cls, systems: List[Union[SatelliteType, str]]):
+        mask = 0
+        for system in systems:
+            if isinstance(system, str):
+                mask |= getattr(cls, system.upper())
+            else:
+                mask |= (1 << int(system))
+        return mask
+
+    @classmethod
+    def bit_mask_to_systems(cls, mask: int):
+        systems = []
+        for system in SatelliteType:
+            if (mask & (1 << int(system))) != 0:
+                systems.append(system)
+        return systems
+
+    @classmethod
+    def bit_mask_to_string(cls, mask: int):
+        systems = cls.bit_mask_to_systems(mask)
+        return ', '.join(str(s) for s in systems)
+
+
 class SignalType(IntEnum):
     UNKNOWN = 0
 
 
+class FrequencyBand(IntEnum):
```

Comment:
For basically all the code in FrequencyBandMaskVal and FrequencyBand, you could probably reuse the same logic as SatelliteTypeMaskVal and SatelliteTypeMask by making some sort of class factory. Probably not worth it for now, but if we add a third bitmask we should probably make a generator.

Reply:
Yeah, I thought the same thing. If/when we add another bitmask, we can generalize this a bit to clean things up.